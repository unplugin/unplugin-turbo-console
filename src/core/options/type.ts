export interface Options {
  /**
   * Add a string prefix to the console output.
   *
   * @see https://utc.yuy1n.io/features/custom-prefix.html
   * @default undefined
   */
  prefix?: string
  /**
   * Add a string suffix to the console output.
   *
   * @see https://utc.yuy1n.io/features/custom-prefix.html
   * @default undefined
   */
  suffix?: string
  /**
   * Launch editor option
   *
   * @see https://utc.yuy1n.io/features/launch-editor.html
   * @default true
   */
  launchEditor?: boolean | LaunchEditorOption
  /**
   * Highlight option
   *
   * @see https://utc.yuy1n.io/features/highlight.html
   * @default true
   */
  highlight?: boolean | HighlightOption
  /**
   * Enable pass logs feature.
   *
   * @see https://utc.yuy1n.io/features/pass-logs.html
   * @default true
   */
  passLogs?: boolean
  /**
   * Enable inspector feature.
   *
   * @see https://utc.yuy1n.io/features/inspector.html
   * @default
   * {
   *   printUrl: true
   * }
   */
  inspector?: boolean | InspectorOption
  /**
   * Custom server option.
   *
   * @default { port: 3070, host: '127.0.0.1' }
   */
  server?: {
    port?: number
    host?: string
  }
}

interface LaunchEditorOption {
  /**
   * The specific editor to launch.
   *
   * @see https://github.com/yyx990803/launch-editor#supported-editors
   * @default undefined
   *
   */
  specifiedEditor?: string
}

interface HighlightOption {
  /**
   * Whether to show extended path name when the file's (or folder's) name contains an element in the array.
   *
   * @see https://utc.yuy1n.io/features/highlight.html#expand-path-file-name
   * @default undefined
   *
   */
  extendedPathFileNames?: string[]
  /**
   * Whether to detect the system theme and adjust the highlight color.
   *
   * @see https://utc.yuy1n.io/features/highlight.html#theme-detect
   * @default false
   *
   */
  themeDetect?: boolean
}

interface InspectorOption {
  /**
   * Print inspector url in the console when server start.
   *
   * @default true
   */
  printUrl?: boolean
}
