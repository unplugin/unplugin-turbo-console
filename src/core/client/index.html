<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Launch Editor</title>
</head>
<style>
  #error-info {
    color: red;
    font-size: 2rem;
  }

  body {
    background-color: white;
  }

  @media (prefers-color-scheme: dark) {
    body {
      background-color: black;
    }
  }
</style>
<body>
  <div id="error-info"></div>
</body>
<script>
 (async () => {
   try {
     const hash = window.location.hash
     const decodeBase64 = window.atob(hash.slice(1))
     await fetch(`/__open-in-editor?file=${encodeURIComponent(decodeBase64)}`)
     window.close()
   }
   catch (error) {
     const errorInfo = document.getElementById('error-info')
     errorInfo.innerText = String(error)
   }
 })()
</script>
</html>
