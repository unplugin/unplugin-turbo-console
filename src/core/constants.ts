export const PLUGIN_NAME = 'unplugin-turbo-console'
export const NUXT_CONFIG_KEY = 'turboConsole'

export const VirtualModules = {
  Init: '~console',
  ThemeDetect: '~console/theme-detect',
  VueDevTools: '~console/vue-devtools',
  ServerInfo: '~console/server-info',
}

export const PLUGIN_SERVER_DEFAULT_PORT = 3070
export const PLUGIN_SERVER_PORT_RANGE: [number, number] = [PLUGIN_SERVER_DEFAULT_PORT, 6000]
