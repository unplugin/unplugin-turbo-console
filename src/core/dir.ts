import { fileURLToPath } from 'node:url'
import { dirname, resolve } from 'pathe'

export const DIR_DIST = typeof __dirname !== 'undefined'
  ? __dirname
  : dirname(fileURLToPath(import.meta.url))

export const CLIENT_DIR = resolve(DIR_DIST, './client')
