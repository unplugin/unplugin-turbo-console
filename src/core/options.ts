import { env } from 'node:process'
import type { Options } from '../types'

const DETAULT_OPTIONS: Options = {
  prefix: '',
  suffix: '',
  disableLaunchEditor: false,
  disableHighlight: false,
  extendedPathFileNames: [],
  port: 3070,
}

export const BUILD_OPTIONS: Options = {
  disableLaunchEditor: true,
  disableHighlight: true,
}

export function resolveOptions(options: Options): Options {
  let resolved = {
    ...DETAULT_OPTIONS,
    ...options,
  }

  if (env.NODE_ENV === 'production') {
    resolved = {
      ...resolved,
      ...BUILD_OPTIONS,
    }
  }

  return resolved
}
