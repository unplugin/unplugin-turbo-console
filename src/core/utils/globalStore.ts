class GlobalStore {
  private data: Record<string, unknown> = {}

  set<T = unknown>(key: string, value: T): void {
    this.data[key] = value
  }

  get<T = unknown>(key: string): T | undefined {
    return this.data[key] as T | undefined
  }

  remove(key: string): void {
    delete this.data[key]
  }

  clear(): void {
    this.data = {}
  }
}

const globalStore = new GlobalStore()
export default globalStore
