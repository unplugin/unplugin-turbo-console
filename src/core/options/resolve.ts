import type { Options } from './type'
import { env } from 'node:process'
import { PLUGIN_SERVER_DEFAULT_PORT } from '../constants'

export function resolveOptions(options: Partial<Options> = {}): Required<Options> {
  const resolved = {
    launchEditor: options.launchEditor ?? true,
    highlight: options.highlight ?? true,
    passLogs: options.passLogs ?? true,
    inspector: options.inspector ?? { printUrl: true },
    server: {
      port: options.server?.port ?? PLUGIN_SERVER_DEFAULT_PORT,
      host: options.server?.host ?? '127.0.0.1',
    },
    prefix: options.prefix ?? '',
    suffix: options.suffix ?? '',
  }

  if (env.NODE_ENV === 'production') {
    resolved.launchEditor = false
    resolved.highlight = false
    resolved.passLogs = false
    resolved.inspector = false
  }

  return resolved
}
