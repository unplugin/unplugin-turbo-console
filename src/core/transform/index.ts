import type { Context } from './../../types'
import MagicString from 'magic-string'
import { parseSync } from 'oxc-parser'
import { genConsoleString, getCompiler, getLineAndColumn, isConsoleExpression, isPluginDisable } from '../utils'
import { walk } from '../utils/walker'
import { compilers } from './compilers'

export async function transform(context: Context) {
  const { code, id, options } = context
  const magicString = new MagicString(code)

  const compiler = await getCompiler(id)

  if (!compiler) {
    return {
      code: magicString.toString(),
      map: magicString.generateMap({
        source: id,
        file: id,
        includeContent: true,
        hires: true,
      }),
    }
  }

  const compileResult = await compilers[compiler](context)

  const oxcParsedResult = parseSync(id, compileResult.script, {
    lang: (compileResult.lang || 'js') as 'js' | 'jsx' | 'ts' | 'tsx' | undefined,
    sourceType: 'module',
  })

  const { program, comments = [] } = oxcParsedResult

  if (isPluginDisable({
    comments,
    originalLine: 1,
    id,
    type: 'top-file',
    compiler,
    script: compileResult.script,
  })) {
    return {
      code: magicString.toString(),
      map: magicString.generateMap({
        source: id,
        file: id,
        includeContent: true,
        hires: true,
      }),
    }
  }

  await walk(program, {
    enter(node) {
      if (isConsoleExpression(node)) {
        const originalExpression = compileResult.script.slice(node.start, node.end)

        if (originalExpression.includes('%c'))
          return false

        const { line, column } = getLineAndColumn(compileResult.script, node.start)

        const originalLine = line + compileResult.line
        const originalColumn = column

        if (isPluginDisable({
          comments,
          originalLine: line,
          id,
          type: 'inline-file',
          compiler,
          script: compileResult.script,
        })) {
          return false
        }

        const consoleMethod = (node as any).callee.property.name
        const args = (node as any).arguments

        const argsStart = args[0].start! + compileResult.offset
        const argsEnd = args[args.length - 1].end! + compileResult.offset
        const argType = args[0].type

        const argsName = magicString.slice(argsStart, argsEnd)
          .toString()
          .replace(/`/g, '')
          .replace(/\n/g, '')
          .replace(/"/g, '')

        const { consoleString, _suffix } = genConsoleString({
          options,
          consoleMethod,
          originalLine,
          originalColumn,
          argType,
          argsName,
          id,
        })

        if (consoleString)
          magicString.appendLeft(argsStart, consoleString)
        if (_suffix)
          magicString.appendRight(argsEnd, `,"${_suffix}"`)
      }
    },
  })

  return {
    code: magicString.toString(),
    map: magicString.generateMap({
      source: id,
      file: id,
      includeContent: true,
      hires: true,
    }),
  }
}
