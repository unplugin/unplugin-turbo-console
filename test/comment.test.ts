import { join } from 'pathe'
import { describe, expect, it, vi } from 'vitest'
import { resolveOptions } from '../src/core/options/resolve'
import { transform } from '../src/core/transform'
import globalStore from '../src/core/utils/globalStore'
import { COMMENT_CURRENT_FILE_VUE, COMMENT_CURRENT_LINE, COMMENT_NEXT_LINE, COMMENT_TOP_FILE, COMMENT_TOP_FILE_SVELTE, COMMENT_TOP_FILE_VUE } from './fixtures/comments'

vi.mock('node:process', () => {
  return {
    cwd: vi.fn(() => join('/', 'mock', 'path')),
    env: {
      NODE_ENV: 'development',
    },

  }
})

const mockFilePathMap = new Map()
mockFilePathMap.set('../../home/runner/comments.js', 'commsx')
mockFilePathMap.set('../../home/runner/comments.vue', 'zsdgg')
mockFilePathMap.set('../../home/runner/comments.svelte', 'xcvgg')

globalStore.set('port', 3070)
globalStore.set('filePathMap', mockFilePathMap)

describe('disable by comments', () => {
  it ('on top of file', async () => {
    COMMENT_TOP_FILE.options = resolveOptions(COMMENT_TOP_FILE.options)

    expect(await transform(COMMENT_TOP_FILE)).toMatchSnapshot()
  })

  it('on top of vue file', async () => {
    COMMENT_TOP_FILE_VUE.options = resolveOptions(COMMENT_TOP_FILE_VUE.options)
    expect(await transform(COMMENT_TOP_FILE_VUE)).toMatchSnapshot()
  })

  it('on top of svelte file', async () => {
    COMMENT_TOP_FILE_SVELTE.options = resolveOptions(COMMENT_TOP_FILE_SVELTE.options)
    expect(await transform(COMMENT_TOP_FILE_SVELTE)).toMatchSnapshot()
  })

  it('on current line', async () => {
    COMMENT_CURRENT_LINE.options = resolveOptions(COMMENT_CURRENT_LINE.options)
    expect(await transform(COMMENT_CURRENT_LINE)).toMatchSnapshot()
  })

  it('on next line', async () => {
    COMMENT_NEXT_LINE.options = resolveOptions(COMMENT_NEXT_LINE.options)
    expect(await transform(COMMENT_NEXT_LINE)).toMatchSnapshot()
  })

  it('on next line vue', async () => {
    COMMENT_CURRENT_FILE_VUE.options = resolveOptions(COMMENT_CURRENT_FILE_VUE.options)
    expect(await transform(COMMENT_CURRENT_FILE_VUE)).toMatchSnapshot()
  })
})
