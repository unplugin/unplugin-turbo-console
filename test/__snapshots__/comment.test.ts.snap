// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`disable by comments > on current line 1`] = `
{
  "code": "console.log("%cðŸš€ comments.jsÂ·1%cðŸ”¦ http://localhost:3070#commsx,1,1",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#111827;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#F7DF1E90;' : '#F7DF1E;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",'hello javascript')
  console.log('hello javascript2') // turbo-console-disable-line
  ",
  "map": SourceMap {
    "file": "comments.js",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,qUAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC/B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAChE,CAAC",
    "names": [],
    "sources": [
      "comments.js",
    ],
    "sourcesContent": [
      "console.log('hello javascript')
  console.log('hello javascript2') // turbo-console-disable-line
  ",
    ],
    "version": 3,
  },
}
`;

exports[`disable by comments > on next line 1`] = `
{
  "code": "console.log("%cðŸš€ comments.jsÂ·1%cðŸ”¦ http://localhost:3070#commsx,1,1",\`padding:2px 5px;border-radius:3px 0 0 3px;margin:0 0 5px 0;color:#111827;background:\${globalThis._UTC_DETECT_DARK && globalThis._UTC_DETECT_DARK() ? '#F7DF1E90;' : '#F7DF1E;'}\`,\`background:#00DC8250;padding:2px 5px;border-radius:0 3px 3px 0;margin:0 0 5px 0;\`,"\\n",'hello javascript')
  // turbo-console-disable-next-line
  console.log('hello javascript2')
  ",
  "map": SourceMap {
    "file": "comments.js",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,qUAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC/B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACpC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAClC,CAAC",
    "names": [],
    "sources": [
      "comments.js",
    ],
    "sourcesContent": [
      "console.log('hello javascript')
  // turbo-console-disable-next-line
  console.log('hello javascript2')
  ",
    ],
    "version": 3,
  },
}
`;

exports[`disable by comments > on next line vue 1`] = `
{
  "code": "<template>
  Hello World
</template>

<script setup lang="ts">
  const msg: string = "Hello World"
  console.log(msg) // turbo-console-disable-line
</script>
",
  "map": SourceMap {
    "file": "comments.vue",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACV,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACb,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACX;AACA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACxB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACnC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAChD,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "comments.vue",
    ],
    "sourcesContent": [
      "<template>
  Hello World
</template>

<script setup lang="ts">
  const msg: string = "Hello World"
  console.log(msg) // turbo-console-disable-line
</script>
",
    ],
    "version": 3,
  },
}
`;

exports[`disable by comments > on top of file 1`] = `
{
  "code": "// turbo-console-disable
  console.log('hello javascript')
  console.log('hello javascript2')
  ",
  "map": SourceMap {
    "file": "comments.js",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACxB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACjC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAClC,CAAC",
    "names": [],
    "sources": [
      "comments.js",
    ],
    "sourcesContent": [
      "// turbo-console-disable
  console.log('hello javascript')
  console.log('hello javascript2')
  ",
    ],
    "version": 3,
  },
}
`;

exports[`disable by comments > on top of svelte file 1`] = `
{
  "code": "<script lang="ts">
// turbo-console-disable
  const a: string = 'Hello'
  console.log(a)
</script>


<h1>Welcome to SvelteKit</h1>
<p>
  Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation
</p>
",
  "map": SourceMap {
    "file": "comments.svelte",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAClB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACxB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC3B,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAChB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACT;AACA;AACA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAC7B,CAAC,CAAC,CAAC;AACH,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACrF,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "comments.svelte",
    ],
    "sourcesContent": [
      "<script lang="ts">
// turbo-console-disable
  const a: string = 'Hello'
  console.log(a)
</script>


<h1>Welcome to SvelteKit</h1>
<p>
  Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation
</p>
",
    ],
    "version": 3,
  },
}
`;

exports[`disable by comments > on top of vue file 1`] = `
{
  "code": "<template>
  Hello World
</template>

<script setup lang="ts">
// turbo-console-disable
  const msg: string = "Hello World"
  console.log(msg)
</script>
",
  "map": SourceMap {
    "file": "comments.vue",
    "mappings": "AAAA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACV,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACb,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACX;AACA,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACxB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACxB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AACnC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;AAClB,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC;",
    "names": [],
    "sources": [
      "comments.vue",
    ],
    "sourcesContent": [
      "<template>
  Hello World
</template>

<script setup lang="ts">
// turbo-console-disable
  const msg: string = "Hello World"
  console.log(msg)
</script>
",
    ],
    "version": 3,
  },
}
`;
